airgun/views/discoveredhosts.py

```diff
@@ -0,0 +1,275 @@
+
+
+class DiscoveredHostsView(BaseLoggedInView, SearchableViewMixin):
+    """Main discovered hosts view"""
+    title = Text("//h1[contains(., 'Discovered Hosts')]")
+    actions = ActionsDropdown("//div[@id='submit_multiple']")
+    table = DiscoveredHostsViewTable(
+        './/table',
+        column_widgets={
+            0: Checkbox(
+                locator=".//input[contains(@id, 'host_ids')]"),
+            'Name': Text("./a"),
+        }
+    )
+    # This view has no welcome message
+    welcome_message = None
+
+    @property
+    def is_displayed(self):
+        return self.browser.wait_for_element(
+            self.title, exception=False) is not None
+
+    def is_searchable(self):
+        """Verify that search procedure can be executed against discovered
+        hosts page. That means that we have search field present.
+
+        Note: When no discovered host exists in the system the search box does
+        not exist in the DOM. Also there is no welcome_message.
+        """
+        # This View is searchable if search field is displayed.
+        return self.searchbox.search_field.is_displayed
+
+
+class DiscoveredHostDetailsTableColumn(TableColumn):
+    """A Table column for Discovered host details Table Row"""
+    def __locator__(self):
+        return self.browser.element(
+            './*[self::th or self::td][{0}]'.format(self.position + 1),
+            parent=self.parent
+        )
+
+
+class DiscoveredHostDetailsTableRow(TableRow):
+    """A Table Row for Discovered host details Table"""
+    Column = DiscoveredHostDetailsTableColumn
+
+
+class DiscoveredHostDetailsTable(Table):
+    """A Table represented by two columns where the property name is the first
+    column and the property value is the second column.
+
+    Example html representation::
+
+        <table>
+            <tbody>
+                <tr>
+                    <th><strong> architecture </strong></th>
+                    <td>x86_64</td>
+                </tr>
+                <tr>
+                    <th><strong> discovery_bootif </strong></th>
+                    <td>52:d2:5f:73:31:e2</td>
+                </tr>
+                <tr>
+                    <th><strong> discovery_bootip</strong></th>
+                    <td>10.8.212.60</td>
+                </tr>
+    """
+    ROWS = './tbody/tr'
+    ROW_AT_INDEX = './tbody/tr[{0}]'
+    # there is no header row elements in this table.
+    HEADER_IN_ROWS = None
+    HEADERS = None
+
+    Row = DiscoveredHostDetailsTableRow
+
+    @property
+    def _is_header_in_body(self):
+        """Explicitly return False as there is no header row in this table."""
+        return False
+
+    @property
+    def headers(self):
+        """Explicitly returns the headers name and value"""
+        return 'name', 'value'
+
+    def read(self):
+        """Transform rows to a dict {property_name: property_value ...}."""
+        properties = super(DiscoveredHostDetailsTable, self).read()
```

Comment:
nitpick: there's shorter py3 version: `properties = super().read()`

Reply:
we can go further more deeper, in py3 implantations, but prefer to select from which class we are using that read.